crates/cargo_utils/src/registry.rs

```diff
@@ -145,3 +172,36 @@ pub fn cargo_home() -> anyhow::Result<PathBuf> {
         .unwrap_or(default_cargo_home);
     Ok(cargo_home)
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn test_get_registry_env_var_name() {
+        assert_eq!(
+            get_registry_env_var_name("my-registry"),
+            "CARGO_REGISTRIES_MY_REGISTRY_INDEX"
+        );
+        assert_eq!(
+            get_registry_env_var_name("my_registry"),
+            "CARGO_REGISTRIES_MY_REGISTRY_INDEX"
+        );
+        assert_eq!(
+            get_registry_env_var_name("registry1"),
+            "CARGO_REGISTRIES_REGISTRY1_INDEX"
+        );
+        assert_eq!(
+            get_registry_env_var_name("UPPERCASE"),
+            "CARGO_REGISTRIES_UPPERCASE_INDEX"
+        );
+        assert_eq!(
+            get_registry_env_var_name("with-special-chars!@#$%^&*()"),
+            "CARGO_REGISTRIES_WITH_SPECIAL_CHARS__INDEX"
```

Comment:
this test case looks wrong, what do you think?

Reply:
Sorry, it appears I was mistaken about multiple non-alphanumeric characters not repeating the `_` replacement.

I just tested a simple name locally like this:

```
CARGO_REGISTRIES_DEMO_INDEX="ssh://git@some.registry/crate-index.git" cargo login --registry demo
```

then, adding multiple dashes like this:
```
CARGO_REGISTRIES_DEMO___TWO_INDEX="ssh://git@some.registry/crate-index.git" cargo login --registry demo---two
```

Also it turns out the characters are much more restricted:

```
❯ CARGO_REGISTRIES_DEMO___TWO_INDEX="ssh://git@some.registry/crate-index.git" cargo login --registry demo@-@two
error: invalid character `@` in registry name: `demo@-@two`, characters must be Unicode XID characters (numbers, `-`, `_`, or most letters)
```

Reply:
Updated now!